runtime/marshaler_registry.go

```diff
@@ -16,42 +16,67 @@ var (
 	defaultMarshaler = &JSONPb{OrigName: true}
 )
 
-// MarshalerForRequest returns the inbound/outbound marshalers for this request.
-// It checks the registry on the ServeMux for the MIME type set by the Content-Type header.
+// MarshalerForRequest returns the outbound marshalers for this request.
+// It checks the registry on the ServeMux for the MIME type set by the Accept header.
+// If it isn't set (or the request Accept is empty), it checks for the Content-Type header.
 // If it isn't set (or the request Content-Type is empty), checks for "*".
 // If there are multiple Content-Type headers set, choose the first one that it can
 // exactly match in the registry.
-// Otherwise, it follows the above logic for "*"/InboundMarshaler/OutboundMarshaler.
-func MarshalerForRequest(mux *ServeMux, r *http.Request) (inbound Marshaler, outbound Marshaler) {
+// Otherwise, it follows the above logic for "*"/OutboundMarshaler.
```

Comment:
I know you only slightly rewrote this, but I'm not sure I understand what this means. If you understand what it means, could you explain? If you are also confused, please just remove this line.